crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,208 @@
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `author_key_digest` in the report body.
+    /// This is the list of disallowed public keys which have signed the signing key of the
+    /// Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `measurement` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `measurement` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub id_key_digest: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub id_key_digest_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Minimum value for `policy.abi_major`.`policy.abi_minor`
+    #[serde(default)]
+    #[serde(deserialize_with = "from_version_string")]
+    pub minimum_abi: Option<Version>,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_policy_string")]
+    pub policy_flags: Option<u8>,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_platform_string")]
+    pub platform_info_flags: Option<PlatformInfoFlags>,
+}
+
+/// This is a work-around for the `semver` crate which requires a patch value. This isn't
+/// applicable here, since the AMD attestation report Policy does not specify this.
+fn from_version_string<'de, D>(deserializer: D) -> Result<Option<Version>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let version = match s.split('.').count() {
+        3 => Version::from_str(s).map_err(|_| Error::custom("invalid version string"))?,
+        2 => {
+            let new_version = format!("{}.0", s);
+            Version::from_str(&new_version).map_err(|_| Error::custom("invalid version string"))?
+        }
+        _ => {
+            return Err(Error::custom("invalid version string"));
+        }
+    };
+
+    Ok(Some(version))
+}
+
+fn from_hex<'de, D>(deserializer: D) -> Result<Option<Vec<Vec<u8>>>, D::Error>
```

Comment:
See above. Besides, this is duplicated in the other crate, so this should be extracted in some shared crate and reused

Reply:
Nathaniel seemed to be against a common crate between these SNP & SGX crates, and it's only one function.

Reply:
Resolved